pepper-apis/src/main/java/org/broadinstitute/ddp/service/actvityinstancebuilder/block/question/ValidationRuleCreator.java

```diff
@@ -0,0 +1,166 @@
+                getHintTitle(ctx, ruleDef),
+                ruleDef.getAllowSave(),
+                ruleDef.getMin(),
+                ruleDef.getMax()
+        );
+    }
+
+    private CompleteRule createCompleteRule(AIBuilderContext ctx, CompleteRuleDef ruleDef) {
+        return new CompleteRule(
+                ruleDef.getRuleId(),
+                resolveRuleMessage(ctx, ruleDef),
+                getHintTitle(ctx, ruleDef),
+                ruleDef.getAllowSave()
+        );
+    }
+
+    private RequiredRule createRequiredRule(AIBuilderContext ctx, RequiredRuleDef ruleDef) {
+        return new RequiredRule(
+                ruleDef.getRuleId(),
+                getHintTitle(ctx, ruleDef),
+                resolveRuleMessage(ctx, ruleDef),
+                ruleDef.getAllowSave()
+        );
+    }
+
+    private AgeRangeRule createAgeRangeRule(AIBuilderContext ctx, AgeRangeRuleDef ruleDef) {
+        return AgeRangeRule.of(
+                ruleDef.getRuleId(),
+                resolveRuleMessage(ctx, ruleDef),
+                getHintTitle(ctx, ruleDef),
+                ruleDef.getAllowSave(),
+                ruleDef.getMinAge(),
+                ruleDef.getMaxAge()
+        );
+    }
+
+    private IntRangeRule createIntRangeRule(AIBuilderContext ctx, IntRangeRuleDef ruleDef) {
+        return IntRangeRule.of(
+                ruleDef.getRuleId(),
+                resolveRuleMessage(ctx, ruleDef),
+                getHintTitle(ctx, ruleDef),
+                ruleDef.getAllowSave(),
+                ruleDef.getMin(),
+                ruleDef.getMax()
+        );
+    }
+
+    private DateRangeRule createDateRangeRule(AIBuilderContext ctx, DateRangeRuleDef ruleDef) {
+        return DateRangeRule.of(
+                ruleDef.getRuleId(),
+                resolveRuleMessage(ctx, ruleDef),
+                getHintTitle(ctx, ruleDef),
+                ruleDef.getAllowSave(),
+                ruleDef.getStartDate(),
+                ruleDef.isUseTodayAsEnd() ? LocalDate.now(ZoneOffset.UTC) : ruleDef.getEndDate()
+        );
+    }
+
+    private DateFieldRequiredRule createDateFieldRequiredRule(AIBuilderContext ctx, DateFieldRequiredRuleDef ruleDef) {
+        return DateFieldRequiredRule.of(
+                ruleDef.getRuleType(),
+                ruleDef.getRuleId(),
+                resolveRuleMessage(ctx, ruleDef),
+                getHintTitle(ctx, ruleDef),
+                ruleDef.getAllowSave()
+        );
+    }
+
+    private NumOptionsSelectedRule createNumOptionsSelectedRule(AIBuilderContext ctx, NumOptionsSelectedRuleDef ruleDef) {
+        return NumOptionsSelectedRule.of(
+                ruleDef.getRuleId(),
+                resolveRuleMessage(ctx, ruleDef),
+                getHintTitle(ctx, ruleDef),
+                ruleDef.getAllowSave(),
+                ruleDef.getMin(),
+                ruleDef.getMax()
+        );
+    }
+
+    private String resolveRuleMessage(AIBuilderContext ctx, RuleDef ruleDef) {
```

Comment:
I don't think we need to do any "resolving" here. If you look in the abstract parent `Rule` class, it will decide whether to use the `correctionHint` or the `defaultMessage`. So we just need to grab the default one from ActivityDefStore and pass it in.